auslib/test/test_db.py

```diff
@@ -2422,6 +2422,32 @@ def testGetRulesMatchingQueryChannelGlobbing(self):
         rules = self._stripNullColumns(rules)
         self.assertEquals(rules, expected)
 
+        # To ensure length of ruleChannel is >=3
+        expected = [dict(rule_id=2, priority=100, backgroundRate=100, channel='r*', update_type='z', data_version=1)]
+        rules = self.rules.getRulesMatchingQuery(
+            dict(name='', product='', version='3.0', channel='releasetest',
+                 buildTarget='', buildID='', locale='', osVersion='', distribution='',
+                 distVersion='', headerArchitecture='',
+                 force=False, queryVersion=3,
+                 ),
+            fallbackChannel='releasetest'
+        )
+        rules = self._stripNullColumns(rules)
+        self.assertNotEquals(rules, expected)
```

Comment:
I don't think you want to be using assertNotEquals. You should be using assertEquals, and ensuring that the data that the rules table is initialized with (https://github.com/mozilla/balrog/blob/a10b66786774262397a51e77931525ea2c0cbbdc/auslib/test/test_db.py#L2297) includes data that lets your tests pass. For example, you will probably need to add an additional rule with channel set to "r*", and then expect that that rule will be returned by getRulesMatchingQuery(). Please let me know if you have trouble with that - I know the tests can be intimidating at first.